### PR TITLE
[Incremental] Reorder IncrementalCompilationTests and extract OutputFileMapCreator

### PR DESCRIPTION
**Pull request by davidungar**
In preparation for adding new incremental import tests, pull out `OutputFileMapCreator`. Also some reordering of `IncrementalCompilationTests`.

### CODE REVIEW

File: Tests/SwiftDriverTests/Helpers/OutputFileMapCreator.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        "dependencies": ".d",
+        "diagnostics": ".dia",
+        "llvm-bc": ".bc",
+        "object": ".o",
+        "swift-dependencies": ".swiftdeps",
+        "swiftmodule": "-partial.swiftmodule"
+      ]
+      .mapValues {"\(derivedData.appending(component: s.basenameWithoutExt))\($0)"}
+    }
+    return Dictionary( uniqueKeysWithValues:
```
Comment (CodaFi) [inline Tests/SwiftDriverTests/Helpers/OutputFileMapCreator.swift:43]: Nit: No extra spaces.

File: Tests/SwiftDriverTests/IncrementalCompilationTests.swift [hunk 2]
```diff
@@ -887,7 +857,7 @@ class CrossModuleIncrementalBuildTests: XCTestCase {
       let main = path.appending(component: "main.swift")
       try localFileSystem.writeFileContents(main) {
         $0 <<< "import MagicKit\n"
-        $0 <<< "castASpell()"
+                $0 <<< "castASpell()"
```
Comment (CodaFi) [inline Tests/SwiftDriverTests/IncrementalCompilationTests.swift:860]: Same here

File: Tests/SwiftDriverTests/Helpers/OutputFileMapCreator.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import TSCBasic
+
+import Foundation
+
+struct OutputFileMapCreator {
+  private let module: String
+  private let inputPaths: [AbsolutePath]
+  private let derivedData: AbsolutePath
+
+  static func write(module: String,
```
Comment (CodaFi) [inline Tests/SwiftDriverTests/Helpers/OutputFileMapCreator.swift:22]: If this is going to be a utility method we ought to hide the member wise initializer

**Review by CodaFi — APPROVED** [review]
Approved modulo whitespace changes.

**Comment by davidungar** [comment]
@swift-ci please test and merge

**Comment by davidungar** [comment]
> Approved modulo whitespace changes.

Thank you.

**Comment by davidungar** [comment]
@swift-ci please test

**Pull request merged**